xgboost_ray/main.py

```diff
@@ -432,7 +442,10 @@ def train(self, rabit_args: List[str], params: Dict[str, Any],
             if num_threads > 0:
                 local_params["num_threads"] = num_threads
             else:
-                local_params["nthread"] = ray.utils.get_num_cpus()
+                try:
+                    local_params["nthread"] = ray._private.utils.get_num_cpus()
+                except AttributeError:
+                    local_params["nthread"] = ray.utils.get_num_cpus()
```

Comment:
what's the purpose of this? maybe we can use `get_resource_ids()` instead of tapping the private call?

Reply:
Ah, the resource IDs is actually much better as we will only assign CPUs we actually requested resources for. Thanks, updated.